### ISSUE TITLE
Allow registration without email

### ISSUE THREAD

**Issue by TheOneWithTheBraid** [nextcloud/registration#273]
### Expected behaviour
It would be great to have a way to allow registration without unnecessary collection of data aka without providing a mail address.

### Actual behaviour
I am required to enter a mail address even though `Let user can register directly without email verification` is activated.

### Server configuration
**Operating system**: Linux 5.9 / Ubuntu 

**Web server:** nginx with traefik and Apache proxies

**Database:** MariaDB

**PHP version:** ask the docker image...

**Nextcloud version:** 20.0.7

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:** docker

**Are you using external storage, if yes which one:** nope

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP

***Does it all matter for a feature request?***

**Comment by nickvergessen** [comment]
It means without verification, not without email.
The thing is accounts without an email can not recover a lost password and similar things.

And well "data collection", this is exactly why everyone runs their Nextcloud, so it's very low on the priority list.

**Comment by TheOneWithTheBraid** [comment]
I see. I actually thought of admins for resetting passwords. In our case, email should be optional hence. I will try to manually work around it by submitting a non-existant random address in the form field by java script.

**Comment by TheOneWithTheBraid** [comment]
For anyone having the same issue:
```js
function makeEmail() {
	var strValues="abcdefghijklmnopqrstuvwxyz1234567890";
	var strEmail = "";
	var strTmp;
	for (var i=0;i<10;i++) {
		strTmp = strValues.charAt(Math.round(strValues.length*Math.random()));
		strEmail = strEmail + strTmp;
	}
	strTmp = "";
	strEmail = strEmail + "@";
	for (var j=0;j<8;j++) {
		strTmp = strValues.charAt(Math.round(strValues.length*Math.random()));
		strEmail = strEmail + strTmp;
	}
	strEmail = strEmail + ".invalid"
	return strEmail;
}

if(window.location.pathName="/apps/registration/") {
    document.querySelector("#email").value = makeEmail();
    var para = document.createElement("p");
    para.innerHTML = '<p style=\"text-shadow: 0 0 2px rgba(0, 0, 0, .4);background: var(--color-background-darker) !important;color: var(--color-main-text) !important;border-radius: 4px;\"><b>Note: </b>You may provide an email address, but for privacy reasons, we <b>do not recommend</b> providing one. Instead, you may use the following random address.</p>';
    var notice = para.children[0];
    document.querySelector("form fieldset").prepend(notice);
}
```

**Comment by nickvergessen** [comment]
Implemented via #301  merged with #364

**Comment by TheOneWithTheBraid** [comment]
@nickvergessen I would like to extend the proposal to an optional setting in the admin area *disallowing* email in registration. For some organizations, this might be a very useful privacy feature.
